### DISCUSSION TITLE
The possibility of adding exoplanets as locations

### DISCUSSION THREAD

**Discussion by Vendrel** [Stellarium/stellarium#3927] [Ideas]
The Context: I recently came across an article discussing the discovery of an exoplanet located in the vicinity of Barnard's Star. Its orbital period is approximately 3 days. It would be fascinating to explore the various celestial phenomena that could be experienced from the surface of such a planet. In due course, telescopic observations will undoubtedly yield empirical data on the planetary rotation periods and axial tilt angles.

So, in the ssystem_major.ini file, for satellites, parent objects (planets) can be specified, but planets' parent object is not specified, though we can define stars (although only one star is defined there: the Sun). You should just add this specification so we could define another stars, then exoplanets orbiting them.

An example of relevant parameters:

```
[proxima_centauri]
absolute_magnitude=15.5
albedo=0.1
color=1.0, 0.7, 0.6
coord_func=proxima_centauri_special
halo=false
name=Proxima Centauri
parent=none
radius=0.144 # In the solar system, it is 14.4% of the Sun's radius
rot_equator_ascending_node=0.0
rot_obliquity=0.0
rot_periode=83.0 # Expressed in days, ~83 days
tex_map=proxima_centauri.png
type=star

[proxima_centauri_b]
absolute_magnitude=30.0 # Exoplanets tend to have high absolute luminosities
albedo=0.3 # Earth-like albedo
atmosphere=1 #Let's assume there is an atmosphere
color=1.0, 0.8, 0.6 # Earth-like color
coord_func=proxima_centauri_b_special
landscape=earth # Assume an earth-like landscape
name=Proxima Centauri b
parent=Proxima Centauri
oblateness=0.002 # Estimate based on Earth data
orbit_visualization_period=11.2 # In days (orbital time approx. 11.2 days)
radius=1.1 # 1.1 times the radius of Earth estimate (Earth-like size)
rot_equator_ascending_node=0.0 # Unknown, so 0
rot_obliquity=23.5 # Make it similar to Earth
rot_periode=24.0 # The length of the day is hours
rot_pole_de=0.0 # Unknown, so 0
rot_pole_de1=0.0 # Unknown, so 0
rot_pole_ra=0.0 # Unknown, so 0
rot_pole_ra1=0.0 # Unknown, so 0
rot_pole_w0=0.0 # Unknown, so 0
rot_pole_w1=0.0 # Unknown, so 0
rot_rotation_offset=0.0 # Unknown, so 0
tex_map=mars.png # Useful landscape
type=planet
```

**Comment by alex-w** [comment]
You need a Celestia :-)

**Reply by gzotti** [reply]
Yes, Stellarium is "limited" to our own solar system, until You change that...

**Reply by Vendrel** [reply]
No, in Celestia, you can't lock yourself to a certain location on an exoplanet, it will turn without you.

**Reply by Vendrel** [reply]
> Yes, Stellarium is "limited" to our own solar system, until You change that...

Yes, this is why this minor improvement would remove this limitation. :)

**Reply by gzotti** [reply]
Minor improvement? Like "just attach wings" to a submarine so it can fly supersonic?

**Reply by Vendrel** [reply]
> Minor improvement? Like "just attach wings" to a submarine so it can fly supersonic?

:D Okay, okay, I just thought exoplanets orbiting other stars could be just like any other location we already have in the application, and their parent stars in some data files, to be addressed just like we address Mars as a location and its parent object as the Sun. But I'm not a developer, just it seemed logical just like that. :D Best! Thanks for the great application!!!

Wait! What if all exoplanets orbit the Sun, technically?

**Reply by 10110111** [reply]
> What if all exoplanets orbit the Sun, technically?

Well, technically, then don't :P 
I guess the limitation is the way star management is organized, but I didn't delve into details.

**Reply by gzotti** [reply]
Our stars are all on an icosaedron. Think "Celestial globe". No distances. For the 3D structure of the galaxy, use Celestia, Space Engine or whatever else you can find. They will probably even show volcanoes or rivers on exoplanets. :-P

**Reply by Vendrel** [reply]
> Our stars are all on an icosaedron. Think "Celestial globe". No distances. For the 3D structure of the galaxy, use Celestia, Space Engine or whatever else you can find. They will probably even show volcanoes or rivers on exoplanets. :-P

I see. :) (Just for the record: Celestia can not be utilized for this usage. You can follow objects and track objects but can't make them your pivot point.)
